app/main.cpp

```diff
@@ -99,50 +99,76 @@ static void setEnvironmentQgisPrefixPath() {
+        QString dstItemPath = dstPath + "/" + info.fileName();
+        if (info.isDir()) {
+            if (!cpDir(srcItemPath, dstItemPath)) {
+                return false;
+            }
+        } else if (info.isFile()) {
+            if (!QFile::copy(srcItemPath, dstItemPath)) {
+                return false;
+            }
+        } else {
+            qDebug() << "Unhandled item" << info.filePath() << "in cpDir";
+        }
+    }
+    return true;
+}
 
-static void expand_assets_data(const QString& qgisDataPath) {
+// Copies resources folder to package folder
+static void expand_pkg_data(const QString& pkgPath) {
 #ifdef ANDROID
   QString assetsBasePath( "assets:" );
   qDebug("assets base path:  %s", assetsBasePath.toLatin1().data());
 
-  QDir qgisDataDir(qgisDataPath);
-
   // make sure all necessary QGIS data are available - unpacked from assets
   //if (!homeDir.exists(qgisDataSubdir)) // do it always on startup - for now
   {
-    QStringList qgisDataFiles;
-    qgisDataFiles << "resources/qgis.db" << "resources/srs.db";
-
-#ifdef QGIS_QUICK_EXPAND_TEST_DATA
-    qgisDataFiles << "background.gpkg";
-#endif
+    QStringList qgisDataFolders;
+    qgisDataFolders << "resources";
 
-    foreach (const QString& dataFile, qgisDataFiles)
+    foreach (const QString& dataFolder, qgisDataFolders)
     {
-      QFile f(assetsBasePath + "/qgis-data/" + dataFile);
-      QString destFilePath = qgisDataPath + "/" + dataFile;
-      QDir destFileDir = QFileInfo(destFilePath).absoluteDir();
-      if (!destFileDir.exists())
-      {
-        bool mkres = qgisDataDir.mkpath(QFileInfo(dataFile).dir().path());
-        qDebug("mkpath [%d] %s", mkres, destFileDir.absolutePath().toLatin1().data());
-      }
-      if (!QFile(destFilePath).exists())
-      {
-        bool res = f.copy(destFilePath);
-        qDebug("copying [%d] %s", res, destFilePath.toLatin1().data());
-        // by default the permissions are set as readable for everyone - and not writable!
-        res = QFile::setPermissions(destFilePath, QFile::ReadUser|QFile::WriteUser|QFile::ReadOwner|QFile::WriteOwner);
-        qDebug("chmod [%d]", res);
-      }
+      cpDir(assetsBasePath + "/qgis-data/" + dataFolder, pkgPath + "/" + dataFolder);
     }
   }
 #else
     Q_UNUSED(qgisDataPath);
 #endif
 }
 
-static void init_qgis(const QString& qgisDataPath)
+static void copy_demo_projects(const QString& dataPath) {
+#ifdef ANDROID
+  QString assetsBasePath( "assets:" );
+  qDebug("assets base path:  %s", assetsBasePath.toLatin1().data());
+
+  QStringList demoProjectNames;
+  demoProjectNames << "survey" << "test_project";
+
+  foreach (const QString& projectName, demoProjectNames)
+  {
+      QDir projectDir(dataPath + "/" + projectName);
+      // if projectFolder exists, do not copy
+      if (projectDir.exists()) continue;
+
+      cpDir(assetsBasePath + "/demo-projects/" + projectName, dataPath + "/" + projectName);
+  }
```

Comment:
Same comment as above - can we have this without for cycle, just cpDir(assetsBasePath + "/demo-projects", dataPath) ?